scientific_library/tvb/simulator/simulator.py

```diff
@@ -450,13 +512,15 @@ def _configure_history(self, initial_conditions):
                     history = numpy.roll(history, shift, axis=0)
                 self.current_step += ic_shape[0] - 1
 
-        if self.integrator.state_variable_boundaries is not None:
-            self.integrator.bound_state(numpy.swapaxes(history, 0, 1))
+        # Make sure that history values are bounded,
+        # and any possible non-state variables are initialized
+        # based on state variable ones (but with no coupling yet...)
+        self._update_and_bound_history(numpy.swapaxes(history, 0, 1))
```

Comment:
`numpy.swapaxes` should return a view but its version dependent, so I'd prefer to avoid it. Why can't this method operate directly on the history buffer as-is?

Reply:
The shape is not correct for the state otherwise. We could also loop across time and modes also...